src/app/shared/components/validate-json-or-yaml/component.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, Input, OnChanges, SimpleChanges} from '@angular/core';
+import {verifyYAML, verifyJSON} from '@shared/utils/common';
+
+@Component({
+  selector: 'km-validate-json-or-yaml',
+  templateUrl: './template.html',
+})
+export class ValidateJsonOrYamlComponent implements OnChanges {
+  @Input() data: string;
+  @Input() isOnlyYAML = false;
+
+  isDataValid = false;
+
+  private _verifyYAML = verifyYAML;
+  private _verifyJSON = verifyJSON;
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.data) {
+      this._setIsDataValid();
+    }
+  }
+
+  private _setIsDataValid(data = this.data) {
```

Comment:
remove -> default parameter method can access direclty

Reply:
I set it this way so that while testing we can just pass custom data in the params of this method instead of doing `component.data = <xyz>` then testing the method

Reply:
refactored